lib/utils.js

```diff
@@ -551,6 +551,27 @@ function resolveReadPreference(options, sources) {
   return readPreference;
 }
 
+function resolveReadPreference(parent, options) {
+  options = options || {};
+  const session = options.session;
+
+  const inheritedReadPreference = parent.readPreference;
+
+  let readPreference;
+  if (options.readPreference) {
+    readPreference = ReadPreference.fromOptions(options);
+  } else if (session && session.inTransaction() && session.transaction.options.readPreference) {
+    // The transaction’s read preference MUST override all other user configurable read preferences.
+    readPreference = session.transaction.options.readPreference;
+  } else if (inheritedReadPreference != null) {
+    readPreference = inheritedReadPreference;
+  } else {
```

Comment:
it seems like this loses the feature of the legacy `resolveReadPreference` that lets a collection default to `db` and having a default read preference. is that intended?

Reply:
Yes, rather than passing in a default read preference to `resolveReadPreference`, I think the default should be inherited from `db`. For example, `command` used to have `resolveReadPreference(options, { db: this, default: ReadPreference.primary }`. Now, the default `readPreference` of `db` is `ReadPreference.primary`, which makes it unnecessary to pass in a default to this function.

Reply:
okay, that makes sense for the default value. I'm still wondering about the case where `parent` is a collection. It seems like the old behavior checked in the following order: `options`, `session`, `collection`, `db`, `default`, and now it only checks `options`, `session`, `parent`. Is this a change in behavior now?

Reply:
I don't think this would be a change in behavior because if `parent` is collection, then it will check `collection.readPreference`. If `collection.s.readPreference` is set, then that value will be used. If not, then `collection.s.db.readPreference` will be checked. If `db.s.readPreference` is set, that value will be used. If not, `ReadPreference.primary` will be used, which is the only value that was previously used as `default`.

Reply:
ah, ok, didn't realize the computed property.